core/src/main/java/org/apache/carbondata/scan/result/iterator/DetailQueryResultIterator.java

```diff
@@ -45,10 +47,28 @@
 
   public DetailQueryResultIterator(List<BlockExecutionInfo> infos, QueryModel queryModel) {
     super(infos, queryModel);
+    this.queryModel = queryModel;
+  }
+
+  private Boolean flag;
+
+  private Long total = 0L;
+
+  private QueryModel queryModel;
+
+  @Override public boolean hasNext() {
+    flag = super.hasNext();
```

Comment:
you can have time and record count variable in AbstractDetailQueryResultIterator and in next you can update the value....and once has next is false you can record the statistics